DSLink.Tests/NodeTests.cs

```diff
@@ -0,0 +1,239 @@
+using Moq;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using System;
+using System.Threading.Tasks;
+
+namespace DSLink.Tests
+{
+    [TestFixture]
+    public class NodeTests
+    {
+        private Mock<AbstractContainer> _mockContainer;
+        private Mock<Connector> _mockConnector;
+        private Mock<Responder> _mockResponder;
+        private Mock<SubscriptionManager> _mockSubManager;
+        private Node _superRootNode;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockContainer = new Mock<AbstractContainer>();
+            _mockConnector = new Mock<Connector>(_mockContainer.Object);
+            _mockResponder = new Mock<Responder>();
+            _mockSubManager = new Mock<SubscriptionManager>(_mockContainer.Object);
+            _superRootNode = new Node("", null, _mockContainer.Object);
+
+            _mockContainer.SetupGet(c => c.Connector).Returns(_mockConnector.Object);
+            _mockContainer.SetupGet(c => c.Responder).Returns(_mockResponder.Object);
+            _mockResponder.SetupGet(r => r.SuperRoot).Returns(_superRootNode);
+            _mockResponder.SetupGet(r => r.SubscriptionManager).Returns(_mockSubManager.Object);
+
+            _mockConnector.Setup(c => c.AddValueUpdateResponse(It.IsAny<JToken>()))
+                .Returns(Task.FromResult(false));
+        }
+
+        [Test]
+        public void TestBannedCharacters()
+        {
+            var bannedChars = Node.BannedChars;
+            foreach (char c in bannedChars)
+            {
+                Assert.Throws<ArgumentException>(() =>
+                {
+                    _superRootNode.CreateChild($"Test{c}").BuildNode();
+                });
+            }
+
+            var multiCharTest = new string(Node.BannedChars);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                _superRootNode.CreateChild(multiCharTest).BuildNode();
+            });
+
+            var noCharTest = "TestNoBannedChars";
+            Assert.DoesNotThrow(() =>
+            {
+                _superRootNode.CreateChild(noCharTest).BuildNode();
+            });
+        }
+
+        [Test]
+        public void TestCreateChildren()
+        {
+            int first = 5; // Number of children of root to create
+            int second = 100; // Number of children to make below each first node.
+            for (int i = 0; i < first; i++)
+            {
+                var firstNode = _superRootNode.CreateChild($"Node{i}").BuildNode();
+                for (int j = 0; j < second; j++)
+                {
+                    firstNode.CreateChild($"Node{j}").BuildNode();
+                }
+            }
+
+            Assert.AreEqual(first, _superRootNode.Children.Count);
+            foreach (var kv in _superRootNode.Children)
+            {
+                Assert.AreEqual(second, kv.Value.Children.Count);
+            }
+        }
+
+        [Test]
+        public void TestRemoveChildren()
```

Comment:
I think create and remove children should have there own test group/fixture/class/whatever. There are multiple cases to be covered if you want to cover them all.

For instance, I don't know if we'll only provide the `RemoveFromParent` method, but the regular 0/1/Many tests could be covered if the parent manages its children.